File: python/KRAKEN/Tests/DW/Arctic/arcticK.py
#==================================================================
#  
#  KRAKEN: Arctic sound speed profile
#  Mexilhoeira Grande, Sex Jul 28 18:01:38 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run arcticK

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readmod import *
from readshd import *

rs = 0.0

print("KRAKEN- Arctic profile")

system("kraken.exe arcticK")
system("fields.exe arcticK < arcticK.flp")

filename = 'arcticK.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )

Modes,bc = readmod('arcticK.mod')

phi = Modes["phi"]
z   = Modes["z"]
k   = Modes["k"]

figure(1)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower',vmin=40,vmax=90)
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,-zs,marker="<",markersize=16,color="k")
xlabel('Range (km)')
ylabel('Depth (m)')
title('KRAKEN - Arctic waveguide')
ylim(Dmax,0)

figure(2)
for i in range(4):  
   rphi = real( phi[ : , i ] ) 
   iphi = imag( phi[ : , i ] )
   thetitle = 'Z_'  + str(i+1) + '(z)' 
   subplot(1,4,i+1)
   plot(rphi,z,iphi,z,'r--')
   ylim(Dmax,0)
   title( thetitle )
   grid(True)
subplot(141)
ylabel('Depth (m)')

show()

print("done.")

File: python/KRAKEN/Tests/DW/Arctic/arcticK.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('KRAKEN - Arctic profile') 

system('kraken.exe arcticK');
system('fields.exe arcticK < arcticK.flp');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'arcticK.shd' );

Modes = read_modes_bin( 'arcticK.mod' );
phi = Modes.phi; phi1 = phi(:,1); phi2 = phi(:,2);  phi3 = phi(:,3); phi4 = phi(:,4);
z   = Modes.z;

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

figure(1), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), caxis([40 90])
h = colorbar;
set( h, 'YDir', 'reverse' );
axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('KRAKEN - Arctic profile, 200 Hz')

figure(2)
subplot(141)
plot(real(phi1),z,imag(phi1),z), axis ij
box on, grid on
subplot(142)
plot(real(phi2),z,imag(phi2),z), axis ij
box on, grid on
subplot(143)
plot(real(phi3),z,imag(phi3),z), axis ij
box on, grid on
subplot(144)
plot(real(phi4),z,imag(phi4),z), axis ij
box on, grid on

disp('done.')
